matrix_client/client.py

```diff
@@ -273,6 +281,17 @@ def login(self, username, password, limit=10, sync=True, device_id=None):
         self.token = response["access_token"]
         self.hs = response["home_server"]
         self.api.token = self.token
+        self.device_id = response["device_id"]
+
+        if self.encryption:
+            if ENCRYPTION_SUPPORT:
```

Comment:
Can we put this check in `__init__` instead? Immediately raise error (rather than logging an error) if instantiating with `encryption=True` and `ENCRYPTION_SUPPORT=False`. Also change `self.encryption` to `self._encryption` to signal that it isn't something that should be changed at runtime, and then we can drop this check from here entirely.

Reply:
Sure, way better design!